packages/data-service/src/data-service.ts

```diff
@@ -2285,6 +2314,48 @@ class DataServiceImpl extends WithLogContext implements DataService {
     return { name, ...normalized };
   }
 
+  @op(mongoLogId(1_001_000_063))
+  async previewUpdate(
+    ns: string,
+    filter: Document,
+    update: Document
+  ): Promise<UpdatePreview> {
+    const coll = this._collection(ns, 'CRUD');
+    const session = this._startSession('CRUD');
+    try {
+      session.startTransaction();
+
+      const docsToPreview = await coll
+        .find(filter, { session })
+        .sort({ _id: 1 })
+        .limit(PREVIEW_SAMPLE)
+        .maxTimeMS(PREVIEW_TIMEOUT)
```

Comment:
Tiny nit: Maybe make these options with default values? That’s a bit easier for testing sometimes